src/Backups/BackupIO_AzureBlobStorage.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "config.h"
+
+#if USE_AZURE_BLOB_STORAGE
+#include <Backups/BackupIO_Default.h>
+#include <Disks/DiskType.h>
+#include <Storages/StorageAzureBlobCluster.h>
+#include <Interpreters/Context_fwd.h>
+
+
+namespace DB
+{
+
+/// Represents a backup stored to Azure
+class BackupReaderAzureBlobStorage : public BackupReaderDefault
+{
+public:
+    BackupReaderAzureBlobStorage(StorageAzureBlob::Configuration configuration_, const ReadSettings & read_settings_, const WriteSettings & write_settings_, const ContextPtr & context_);
+    ~BackupReaderAzureBlobStorage() override;
+
+    bool fileExists(const String & file_name) override;
+    UInt64 getFileSize(const String & file_name) override;
+    std::unique_ptr<SeekableReadBuffer> readFile(const String & file_name) override;
+
+    void copyFileToDisk(const String & path_in_backup, size_t file_size, bool encrypted_in_backup,
+                        DiskPtr destination_disk, const String & destination_path, WriteMode write_mode) override;
+
+private:
+    const DataSourceDescription data_source_description;
+    MultiVersion<Azure::Storage::Blobs::BlobContainerClient> client;
+    StorageAzureBlob::Configuration configuration;
+    std::unique_ptr<AzureObjectStorage> object_storage;
+    MultiVersion<AzureObjectStorageSettings> settings;
```

Comment:
Once again, why is it a `MultiVersion`? The `MultiVersion` class is designed to keep multiple versions of an object. You never call `settings.set()` or `client.set()`, those objects have only one version of themselves. So it shouldn't be `MultiVersion`. Just `std::shared_ptr<const AzureObjectStorageSettings> settings` is enough here.

Reply:
addressed it